src/main/scala/org/kframework/kore/Interface.scala

```diff
@@ -0,0 +1,297 @@
+package org.kframework.kore
+
+trait Definition {
+  def modules: Seq[Module]
+  def att: Attributes
+}
+
+object Definition {
+  def unapply(arg: Definition): Option[(Seq[Module], Attributes)] = Some(arg.modules, arg.att)
+}
+
+trait Module {
+  def name: ModuleName
+  def sentences: Seq[Sentence]
+  def att: Attributes
+}
+
+object Module {
+  def unapply(arg: Module): Option[(ModuleName, Seq[Sentence], Attributes)] = Some(arg.name, arg.sentences, arg.att)
+}
+
+trait Sentence {
+  def att: Attributes
+}
+
+trait Import extends Sentence {
+  def name: ModuleName
+}
+
+object Import {
+  def unapply(arg: Import): Option[(ModuleName, Attributes)] = Some(arg.name, arg.att)
+}
+
+trait SortDeclaration extends Sentence {
+  def sort: Sort
+}
+
+object SortDeclaration {
+  def unapply(arg: SortDeclaration): Option[(Sort, Attributes)] = Some(arg.sort, arg.att)
+}
+
+trait SymbolDeclaration extends Sentence {
+  def sort: Sort
+  def symbol: Symbol
+  def args: Seq[Sort]
+}
+
+object SymbolDeclaration {
+  def unapply(arg: SymbolDeclaration): Option[(Sort, Symbol, Seq[Sort], Attributes)] = Some(arg.sort, arg.symbol, arg.args, arg.att)
+}
+
+
+trait Rule extends Sentence {
+  def _1: Pattern
+}
+
+object Rule {
+  def unapply(arg: Rule): Option[(Pattern, Attributes)] = Some(arg._1, arg.att)
+}
+
+trait Axiom extends Sentence {
+  def _1: Pattern
+}
+
+object Axiom {
+  def unapply(arg: Axiom): Option[(Pattern, Attributes)] = Some(arg._1, arg.att)
+}
+
+trait Attributes {
+  def att: Seq[Pattern]
+}
+
+object Attributes {
+  def unapply(arg: Attributes): Option[Seq[Pattern]] = Some(arg.att)
+}
+
+
+trait Pattern
+
+trait Variable extends Pattern
+
+trait SortedVariable extends Variable {
+  def name: Name
+  def sort: Sort
+}
+
+object SortedVariable {
+  def unapply(arg: SortedVariable): Option[(Name, Sort)] = Some(arg.name, arg.sort)
+}
+
+trait Application extends Pattern {
+  def symbol: Symbol
+  def args: Seq[Pattern]
+}
+
+object Application {
+  def unapply(arg: Application): Option[(Symbol, Seq[Pattern])] = Some(arg.symbol, arg.args)
+}
+
+trait DomainValue extends Pattern {
+  def symbol: Symbol
+  def value: Value
+}
+
+object DomainValue {
+  def unapply(arg: DomainValue): Option[(Symbol, Value)] = Some(arg.symbol, arg.value)
+}
+
+trait Top extends Pattern
+
+object Top {
+  def unapply(arg: Top): Boolean = true
```

Comment:
The unapply only denotes match success/failure. 
So, this is scala way of allowing 
```Scala
s match {
  case Top() =>... //Success
}
```